app/models/piece.rb

```diff
@@ -1,9 +1,76 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  PAWN = 'Pawn'.freeze
+  ROOK = 'Rook'.freeze
+  KNIGHT = 'Knight'.freeze
+  BISHOP = 'Bishop'.freeze
+  QUEEN = 'Queen'.freeze
+  KING = 'King'.freeze
+
+  def obstructed?(destination_x, destination_y)
+    return true if self.invalid?(destination_x, destination_y)
+    return true if self.horizontal_or_vertical_obstruction?(destination_x, destination_y)
+    return true if self.diagonal_obstruction?(destination_x, destination_y)
+    false
+  end
+
+  def horizontal_or_vertical?(destination_x, destination_y)
+    return destination_x == self.x_position || destination_y == self.y_position
+  end
+
+  def diagonal?(destination_x, destination_y)
+    # in order for the move to be diagonal, the piece must be moving by the same distance both horizontally and vertically.
+    return true if (destination_x - self.x_position).abs == (destination_y - self.y_position).abs
+  end
+
+  def invalid?(destination_x, destination_y)
+    return true if destination_x < 1 || destination_x > 8 || destination_y < 1 || destination_y > 8
+  end
+
+  def horizontal_or_vertical_obstruction?(destination_x, destination_y)
+    return false unless self.horizontal_or_vertical?(destination_x, destination_y)
```

Comment:
`self.horizontal_or_vertical?` and `horizontal_or_vertical?` will do the same thing.

`self` is assumed when calling a method from inside a class. So you can remove `self` everywhere (I'm guessing that rubocop is probably complaining about this too)